Fails to link on centos 7.4 with -static

```
cc -pthread -std=c99 -static -Wno-everything -O3 -g0 -I /usr/local/include dns.c -c -o dns.o
g++ -pthread -static -Wno-everything -O3 -g0 -I /usr/local/include -Wno-invalid-offsetof -c -o bitcoin.o bitcoin.cpp
g++ -pthread -static -Wno-everything -O3 -g0 -I /usr/local/include -Wno-invalid-offsetof -c -o netbase.o netbase.cpp
g++ -pthread -static -Wno-everything -O3 -g0 -I /usr/local/include -Wno-invalid-offsetof -c -o protocol.o protocol.cpp
g++ -pthread -static -Wno-everything -O3 -g0 -I /usr/local/include -Wno-invalid-offsetof -c -o db.o db.cpp
g++ -pthread -static -Wno-everything -O3 -g0 -I /usr/local/include -Wno-invalid-offsetof -c -o main.o main.cpp
g++ -pthread -static -Wno-everything -O3 -g0 -I /usr/local/include -Wno-invalid-offsetof -c -o util.o util.cpp
g++ -pthread -static -Wno-everything -O3 -g0 -I /usr/local/include -o dnsseed dns.o bitcoin.o netbase.o protocol.o db.o main.o util.o blake256.o -lcrypto
/bin/ld: cannot find -lcrypto
/bin/ld: cannot find -lstdc++
/bin/ld: cannot find -lm
/bin/ld: cannot find -lpthread
/bin/ld: cannot find -lc
collect2: error: ld returned 1 exit status
make: *** [dnsseed] Error 1
```

I don't think I'm missing the packages with the static archives to link this statically, but it works without -static.

--- a/Makefile
+++ b/Makefile
@@ -1,5 +1,6 @@
 INCDIR ?= -I /usr/local/include
-CXXFLAGS = -static -Wno-everything -O3 -g0 ${INCDIR}
+STATIC ?=
+CXXFLAGS = $(STATIC) -Wno-everything -O3 -g0 ${INCDIR}
 LDFLAGS = $(CXXFLAGS)
 
 # OpenBSD and Bitrig require egcc and eg++

--- a/README.md
+++ b/README.md
@@ -43,6 +43,9 @@ using -m.
 
 This will produce the `dnsseed` binary executable.
 
+To build statically (useful for running on minimal debian servers that may not have all of the required libs installed):
+
+    STATIC="-static" make
 
 ### Running as non-root
 

